iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/UnusedVariablesCheckTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.checks;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.arm.parser.BicepParser;
+import org.sonar.iac.arm.symbols.Symbol;
+import org.sonar.iac.arm.symbols.SymbolTable;
+import org.sonar.iac.arm.symbols.Usage;
+import org.sonar.iac.arm.tree.api.File;
+import org.sonar.iac.arm.visitors.ArmSymbolVisitor;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.testing.Verifier;
+
+import static org.mockito.Mockito.mock;
+import static org.sonar.iac.common.testing.Verifier.issue;
+
+class UnusedVariablesCheckTest {
+
+  private static final UnusedVariablesCheck CHECK = new UnusedVariablesCheck();
+
+  @Test
+  void shouldRaiseOnUnusedVariablesJson() {
+    ArmVerifier.verify("UnusedVariablesCheckTest/unusedVariable.json", CHECK,
+      issue(5, 4, 5, 25, "Remove the unused variable \"unusedInParamString\"."),
+      issue(6, 4, 6, 23, "Remove the unused variable \"unusedInParamBool\"."),
+      issue(7, 4, 7, 22, "Remove the unused variable \"unusedInParamInt\"."),
+      issue(8, 4, 8, 24, "Remove the unused variable \"unusedInParamArray\"."),
+      issue(11, 4, 11, 25, "Remove the unused variable \"unusedInParamObject\"."),
+      issue(14, 4, 14, 20, "Remove the unused variable \"unusedVariable\"."));
+  }
+
+  @Test
+  void shouldNotRaiseAnyIssuesAsAVariableUsageIsUnresolvable() {
+    ArmVerifier.verifyNoIssue("UnusedVariablesCheckTest/containsUnresolvableVariable.json", CHECK);
+  }
+
+  @Test
+  void shouldRaiseOnUnusedVariablesBicep() {
+    BicepVerifier.verify("UnusedVariablesCheckTest/unusedVariable.bicep", CHECK);
+  }
+
+  @Test
+  void shouldNotRaiseAnythingWhenNoAssignmentDetected() {
+    BicepParser parser = BicepParser.create();
+    Path path = Paths.get("src", "test", "resources", "checks").resolve("UnusedVariablesCheckTest/unusedVariable.bicep");
```

Comment:
I don't like inconsistency between how the path to the file is passed here and in other checks.
I would suggest creating a small method in `BicepVerifier` (or a separate class `BicepSymbolVerifier`) to get the symbol table and hide such details on how to create the symbol table.